### DISCUSSION TITLE
FileNotFoundError: [WinError 2] 系统找不到指定的文件，怎么办

### DISCUSSION THREAD

**Discussion by bihaiqinkong** [openai/whisper#1161] [Q&A]
PS C:\Users\liuxin> whisper audio.mp3
E:\python 3.10\lib\site-packages\whisper\transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
  warnings.warn("FP16 is not supported on CPU; using FP32 instead")
Traceback (most recent call last):
  File "E:\python 3.10\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "E:\python 3.10\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "E:\python 3.10\Scripts\whisper.exe\__main__.py", line 7, in <module>
  File "E:\python 3.10\lib\site-packages\whisper\transcribe.py", line 437, in cli
    result = transcribe(model, audio_path, temperature=temperature, **args)
  File "E:\python 3.10\lib\site-packages\whisper\transcribe.py", line 121, in transcribe
    mel = log_mel_spectrogram(audio, padding=N_SAMPLES)
  File "E:\python 3.10\lib\site-packages\whisper\audio.py", line 130, in log_mel_spectrogram
    audio = load_audio(audio)
  File "E:\python 3.10\lib\site-packages\whisper\audio.py", line 46, in load_audio
    ffmpeg.input(file, threads=0)
  File "E:\python 3.10\lib\site-packages\ffmpeg\_run.py", line 313, in run
    process = run_async(
  File "E:\python 3.10\lib\site-packages\ffmpeg\_run.py", line 284, in run_async
    return subprocess.Popen(
  File "E:\python 3.10\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "E:\python 3.10\lib\subprocess.py", line 1440, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] 系统找不到指定的文件。

**Comment by Purfview** [comment]
That means that ffmpeg is not found.

**Reply by Darcy-Lee** [reply]
Really kind of you, thank you

**Reply by Robin0315** [reply]
I also encountered this problem, how can I solve it? I have downloaded ffmpeg and I see that ffmpeg is already in the site-packages folder.

**Reply by heimoshuiyu** [reply]
ffmpeg is not a Python package, it is an external program. You need to download it from the official website and set environment variables so that the system can find it.

For Linux, simply run `apt install ffmpeg` or `yum install ffmpeg` or whatever package manager you are using.

For Windows, 

1. Download ffmpeg for Windows from the official website.
2. Extract the zip file to a suitable location on your computer, such as C:\ffmpeg.
3. Open the Start menu and right-click on "Computer" or "This PC" and click on "Properties".
4. Click on "Advanced system settings" on the left sidebar.
5. Click on the "Environment Variables" button at the bottom.
6. In the "System variables" section, scroll down and select the "Path" variable and click on "Edit".
7. Click on "New" and enter the path to the ffmpeg bin folder, such as "C:\ffmpeg\bin".
8. Click on "OK" for all the windows to close them.

**Reply by Robin0315** [reply]
Thanks

**Reply by IamKJLUM** [reply]
Thanks man!

**Reply by pythonchangemylife** [reply]
> That means that ffmpeg is not found.

I am sure I have install ffmpeg,because I can run moviepy.But when I running whisper,it return "FileNotFoundError: [WinError 2] 系统找不到指定的文件。

"

**Reply by Purfview** [reply]
> I am sure I have install ffmpeg,

You don't "install" it, you need to actually download it.
If you are on Windows you can try ready to run Faster-Whisper [it doesn't need ffmpeg] from there: https://github.com/Purfview/whisper-standalone-win

**Reply by megrez-alioth** [reply]
I have really download ffmpeg but there's no "bin" in the bag so I just don't know how to enter that in to PATH

**Reply by rizi960** [reply]
到ffmpeg官网下载一个realse版本，解压缩后，把bin的路径添加到系统环境变量，再dos里执行ffmpeg看是否有版本信息，有则ffmpeg没问题

**Comment by xiaoyaojiushao** [comment]
It seems you are using windows, so you need add ffmpeg to environment variable path, then restart cmd window, then will be OK!
